crates/corro-types/src/channel.rs

```diff
@@ -0,0 +1,107 @@
+use metrics::{gauge, Gauge};
+use std::{
+    fmt::{self, Debug, Formatter},
+    time::Duration,
+};
+use tokio::sync::mpsc::{
+    channel,
+    error::{SendError, SendTimeoutError, TryRecvError, TrySendError},
+    Receiver, Sender,
+};
+
+/// A tokio channel wrapper sender that tracks occupancy
+pub struct CorroSender<T> {
+    gauge: Gauge,
+    inner: Sender<T>,
+}
+
+impl<T> Clone for CorroSender<T> {
+    fn clone(&self) -> Self {
+        Self {
+            gauge: self.gauge.clone(),
+            inner: self.inner.clone(),
+        }
+    }
+}
+
+impl<T: Debug> Debug for CorroSender<T> {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        self.inner.fmt(f)
+    }
+}
+
+/// A tokio channel wrapper receiver that tracks occupancy
+pub struct CorroReceiver<T> {
+    gauge: Gauge,
+    inner: Receiver<T>,
+}
+
+impl<T: Debug> Debug for CorroReceiver<T> {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        self.inner.fmt(f)
+    }
+}
+
+/// Create a bounded channel which tracks occupancy with a label
+pub fn bounded<T>(max: usize, label: &'static str) -> (CorroSender<T>, CorroReceiver<T>) {
+    let gauge = gauge!("corro.runtime.channel_occupancy", "max_capacity" => max.to_string(), "label" => label);
```

Comment:
Label `label` called "label" eh? I assume that's fine.

What's the `max_capacity` for? If it's for charting, it might be more efficient to use a separate metric like `corro.runtime.channel.max_capacity` and set it once when initializing.

Do we care for tracking the "occupancy" vs how much is left? If it was instead called `channel.capacity`, we could more easily alert when it reaches zero.

Reply:
Yeah, we have some channels that use different sizes. I thought it would be useful to see how much of a channel is actually in use vs just the raw usage number. Happy to change any of the actual metric names (I'm new to this stuff anyway)